google/cloud/logging_v2/logger.py

```diff
@@ -48,19 +49,23 @@ class Logger(object):
     See https://cloud.google.com/logging/docs/reference/v2/rest/v2/projects.logs
     """
 
-    def __init__(self, name, client, *, labels=None, resource=_GLOBAL_RESOURCE):
+    def __init__(self, name, client, *, labels=None, resource=None):
         """
         Args:
             name (str): The name of the logger.
             client (~logging_v2.client.Client):
                 A client which holds credentials and project configuration
                 for the logger (which requires a project).
-            resource (~logging_v2.Resource): a monitored resource object
-                representing the resource the code was run on.
+            resource (Optional[~logging_v2.Resource]): a monitored resource object
+                representing the resource the code was run on. If not given, will
+                be inferred from the environment.
             labels (Optional[dict]): Mapping of default labels for entries written
                 via this logger.
 
         """
+        if not resource:
+            # infer the correct monitored resource from the local environment
+            resource = detect_resource(client.project)
```

Comment:
will this default to "global" in case the environment wasn't detected? (i.e. the code is running in dataflow job)

Reply:
Yes, global is the fallback for `detect_resource`